cypress/integration/creatorOnboardingFlows/creatorSettings.spec.js

```diff
@@ -100,7 +102,7 @@ describe('Creator Settings Page', () => {
     it('should show an error when the contrast ratio of a brand color is too low', () => {
       const lowContrastColor = '#a6e8a6';
 
-      cy.findByLabelText(/^Brand color/)
+      cy.findByRole('textbox', { name: /^Brand color/ })
```

Comment:
I love these shifts from `findByLabelText` to `findByRole`.  Wondering if it's more durable (and possible) to instead of using the label name to use the  `name` attribute of the `input` element?  

Definitely not a blocker, but I look to the `input` element's `name` attribute as more durable than it's label text.

Reply:
It would technically be possible to target by the `name` attribute, _but_ then we would be testing an implementation detail. We use these `findByRole` queries provided by `testing-library` precisely because it helps us test closer to how our users perceive and interact with the app - they're unlikely to perceive individual attributes on the input, or e.g. a CSS classname, but they should perceive the label text. It helps us test accessibility too, as if the label ever accidentally becomes disassociated from the input, we'll start seeing errors in the E2E tests, even though visually everything will look and work correctly.

There's quite a few bonuses in using `findByRole` over other query types wherever we can - [check out this bit of the developer docs](https://developers.forem.com/tests/e2e-tests#finding-elements).

Reply:
That's an interesting framing perspective.  "Your" approach favors the text the user will see.  "My" approach favors an implementation detail to make the test more "durable".

And as I consider the implications, I think (especially for integration/UI tests) it's best to favor the user.

Reply:
It's definitely a little bit of a trade off - like you say, some properties are less subject to change than the accessible name (e.g. IDs), but for integration tests it probably makes sense for a test to fail/need updated when we change something user-facing like a label. 

There's definitely been a bit of a shift in frontend integration tests in the last few years to focus more on these user-perceivable selectors - I wrote a little bit about it a while ago (albeit through an a11y lens) - [Accessibility-First Integration Tests with React Testing Library](https://dev.to/s_aitchison/accessibility-first-integration-tests-with-react-testing-library-5fk)